DOC Add User guide page on assembling (#576)
skrub-data/skrub: doc/assembling.rst

====================================
Assembling: joining multiple tables
====================================

.. currentmodule:: skrub

Assembling is the process of collecting and joining together tables. 
Good analytics requires including as much information as possible,
often from different sources.
skrub allows you to join tables on keys of different types
(string, numerical, datetime) with imprecise correspondance.

Fuzzy joining tables
---------------------

> Joining two dataframes is hard as the corresponding keys are often different.

Reviewer: ```suggestion
Joining two dataframes can be hard as the corresponding keys may be different.
```

> The :func:`fuzzy_join` uses similarity scores to join tables on one or more

Reviewer: ```suggestion
The :func:`fuzzy_join` uses similarities in entries to join tables on one or more
```

> related columns. Furthermore, it adapts the fuzzy matching based on the
> column type (string, numerical or datetime).

Reviewer: ```suggestion
related columns. Furthermore, it choose the type of fuzzy matching used
based on the column type (string, numerical or datetime).
```

> Using the similarity score, bad matches are easily singled out and can be
> dropped or replaced.

Reviewer: ```suggestion
It also outputs a similarity score, to single out bad matches, so that they can be
dropped or replaced.
```

> In sum, equivalent to pandas.merge, the :func:`fuzzy_join` 

Reviewer: We need to use intersphinx, to link to pandas. It's set up in https://github.com/skrub-data/skrub/blob/main/doc/conf.py#L277 . It should work
```suggestion
In sum, equivalent to :func:`pandas.merge`, the :func:`fuzzy_join` 
```

has no need for pre-cleaning.

Feature augmentation for machine learning

> -----------------------------------------

Reviewer: Here also I would also stress that this is a scikit-learn transformer, and thus can be used in a pipeline.

Author: In the title as well? I added a paragraph below..

Joining is pretty straigthforward for two tables: you only need to identify
the common key.
However, for more complex analysis, merging multiple tables is necessary.
skrub provides the :class:`FeatureAugmenter` as a convenient solution:
multiple fuzzy joins can be performed at the same time, given a set of 
input tables and key columns.

An advantage is the scikit-learn compatibility of this class:
easily introduced into machine learning pipelines.

Going further: embeddings for better analytics
----------------------------------------------

> 

Reviewer: TODO :)
